File: hdl-simulator/src/main/java/hardwaresimulator/HardwareSimulator.java
package hardwaresimulator;

import java.util.List;

interface HardwareSimulator {
    void loadChipDefinitions(String path);
    void loadChipDefinitions(List<String> path);

    void loadChip(String name);

    void setInput(String name, List<Boolean> values);

    List<Boolean> getOutput(String gateName);
}

File: hdl-simulator/src/main/kotlin/hardwaresimulator/HardwareSimulator.kt
package hardwaresimulator

interface HardwareSimulator {
    fun loadChipDefinitions(vararg paths: String)

    fun loadChip(name: String)

    fun setInputs(vararg inputValues: Pair<String, List<Boolean>>)

    fun getOutput(gateName: String): List<Boolean>
}